### ISSUE TITLE
InsufficientCapabilitiesException error while updating existent IAMServiceAccounts

### ISSUE THREAD

**Issue by aston-r** [weaveworks/eksctl#3528]
Hi,

STR:
- define IAMServiceAccount and run `eksctl create iamserviceaccount --config-file cluster.yaml`
```
iam:
  withOIDC: true
  serviceRoleARN: ""
  serviceAccounts:
    - metadata:
        name: config-service
        namespace: configs
      attachPolicyARNs:
        - arn:aws:iam::aws:policy/AmazonS3ReadOnlyAccess
      roleName: testing-us-east-1-config-service-IRSARole
```
- add one more policy and run `eksctl update iamserviceaccount --config-file cluster.yaml`

```
iam:
  withOIDC: true
  serviceRoleARN: ""
  serviceAccounts:
    - metadata:
        name: config-service
        namespace: configs
      attachPolicyARNs:
        - arn:aws:iam::aws:policy/AmazonS3ReadOnlyAccess
        - arn:aws:iam::aws:policy/AmazonEC2ReadOnlyAccess        
      roleName: testing-us-east-1-config-service-IRSARole
```

Here is an error:
```
2021-04-06 07:52:21 [ℹ]  eksctl version 0.41.0
2021-04-06 07:52:21 [ℹ]  using region us-east-1
2021-04-06 07:52:21 [ℹ]  1 parallel tasks: { 1 task: { update IAMServiceAccount configs/config-service }}
2021-04-06 07:52:21 [ℹ]  updating policies for IAMServiceAccount configs/config-service
2021-04-06 07:53:15 [ℹ]  1 error(s) occurred and IAM Role stacks haven't been updated properly, you may wish to check CloudFormation console
2021-04-06 07:53:15 [✖]  creating ChangeSet "updating-policy" for stack "eksctl-testing-addon-iamserviceaccount-configs-config-service": InsufficientCapabilitiesException: Requires capabilities : [CAPABILITY_NAMED_IAM]
status code: 400, request id: 52f89364-1fbe-46d3-8261-02548008b676
Error: failed to create iamserviceaccount(s)
```

I think it is due to the fact that I have static roleName for the service account, since according to AWS documentation (https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-role.html#cfn-iam-role-rolename):

> If you specify a name(roleName), you must specify the CAPABILITY_NAMED_IAM value to acknowledge your template's capabilities. For more information, see Acknowledging IAM Resources in AWS CloudFormation Templates.

**Comment by nmaupu** [comment]
Same problem here, thanks for posting this !

**Comment by aclevername** [comment]
Thanks for opening the issue @aston-r, I will take a look :smile:

**Comment by aclevername** [comment]
fix should be out in todays `0.45.0-rc.0`
